core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -81,40 +81,15 @@ class SparkHadoopUtil extends Logging {
    * configuration.
    */
   def appendS3AndSparkHadoopConfigurations(conf: SparkConf, hadoopConf: Configuration): Unit = {
```

Comment:
It's so SparkSubmit can create a `Configuration` the same way the rest of the code does, but without initializing the internal state in the `SparkHadoopUtil` class (not the object).